packages/esm-outpatient-app/src/transition-queue-entry/transition-queue-entry-dialog.component.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useCallback, useState } from 'react';
+import { MappedVisitQueueEntry, updateQueueEntry } from '../active-visits/active-visits-table.resource';
+import { useTranslation } from 'react-i18next';
+import {
+  ConfigObject,
+  ExtensionSlot,
+  formatDatetime,
+  navigate,
+  parseDate,
+  showNotification,
+  showToast,
+  toDateObjectStrict,
+  toOmrsIsoString,
+  useConfig,
+} from '@openmrs/esm-framework';
+import { Button, ModalBody, ModalFooter, ModalHeader } from '@carbon/react';
+import { mutate } from 'swr';
+import styles from './transition-queue-entry-dialog.scss';
+import dayjs from 'dayjs';
+import { usePatientAppointments } from '../queue-patient-linelists/queue-linelist.resource';
+import { usePastVisits } from '../past-visit/past-visit.resource';
+interface TransitionQueueEntryProps {
+  queueEntry: MappedVisitQueueEntry;
+  closeModal: () => void;
+}
+const TransitionQueueEntry: React.FC<TransitionQueueEntryProps> = ({ queueEntry, closeModal }) => {
+  const { t } = useTranslation();
+
+  const [selectedQueueLocation, setSelectedQueueLocation] = useState(queueEntry?.queueLocation);
+  const config = useConfig() as ConfigObject;
+  const defaultTransitionStatus = config.concepts.defaultTransitionStatus;
+
+  const preferredIdentifiers = queueEntry?.identifiers.filter((identifier) =>
+    config.defaultIdentifierTypes.includes(identifier?.identifierType?.uuid),
+  );
+
+  const startDate = dayjs(new Date().toISOString()).subtract(6, 'month').toISOString();
+  const { upcomingAppointment, isLoading } = usePatientAppointments(
+    queueEntry?.patientUuid,
+    startDate,
+    new AbortController(),
+  );
+  const { visits, isLoading: loading } = usePastVisits(queueEntry?.patientUuid);
+
+  const launchEditPriorityModal = useCallback(() => {
+    const endedAt = toDateObjectStrict(toOmrsIsoString(new Date()));
+    updateQueueEntry(
+      queueEntry?.visitUuid,
+      queueEntry?.queueUuid,
+      queueEntry?.queueUuid,
+      queueEntry?.queueEntryUuid,
+      queueEntry?.patientUuid,
+      queueEntry?.priorityUuid,
```

Comment:
we can pass `queueEntry` as one object instead of passing individual properties
```suggestion
    quertEntry:queryEntry
```